src/server.js

```diff
@@ -0,0 +1,276 @@
+/**
+ * KumaScript HTTP service
+ *
+ * This file defines the Server class, an Express-based HTTP server
+ * for rendering KumaScript macros within HTML documents.
+ *
+ * The endpoint for rendering documents is a POST to /doc/, with
+ * environment variables encoded in request headers.
+ *
+ * The server also supports these endpoints:
+ *
+ * - GET /macros/ returns a list of defined macros
+ * - GET /revision/ returns the git revision of this version of KumaScript
+ * - GET /healthz/ returns a 204 status if the server is running
+ * - GET /readiness/ returns 204 if the server is ready for requests
+ *
+ * @prettier
+ */
+const path = require('path');
+const url = require('url');
+
+const express = require('express');
+const morgan = require('morgan');
+const request = require('request');
+const winston = require('winston');
+
+const config = require('./config.js');
+const firelogger = require('./firelogger.js');
+const Templates = require('./templates.js');
+const render = require('./render.js');
+
+// TODO: figure out the newrelic config details and re-enable before deploying
+//var newrelic_conf = ks_conf.nconf.get('newrelic');
+//if (newrelic_conf && newrelic_conf.license_key) require('newrelic');
+
+class Server {
+    /**
+     * Build the service, but do not listen yet.
+     */
+    constructor() {
+        const app = (this.app = express());
+
+        // This will raise an error if the macros are not found
+        this.templates = new Templates(config.macrosDirectory);
+
+        // Configure a logger that pipes to the winston logger.
+        if (config.logging) {
+            var logger = morgan(
+                ':remote-addr - - [:date] ":method :url ' +
+                    'HTTP/:http-version" :status :res[content-length] ' +
+                    '":referrer" ":user-agent" :response-time',
+                {
+                    stream: {
+                        write: function(s) {
+                            winston.info(s.trim(), {
+                                source: 'server',
+                                pid: process.pid
+                            });
+                        }
+                    }
+                }
+            );
+
+            app.use(logger);
+        }
+
+        // Define a response.log.error() function that allows us to send
+        // KumaScript error messages back to Kuma in HTTP response headers
+        app.use(firelogger({ logger: config.logging ? winston.info : null }));
```

Comment:
In my reading of the `firelogger` code, it doesn't use the `logger` attribute of the options object passed into it, unless I'm missing something? I would vote for dumping `winston` here, which I think is the final use of `winston`, so we could dump it as a dependency as well, and simplify to this:
```
app.use(firelogger());
```

Reply:
I suppose I could probably just pass `console.log` instead of `winston.info` to retain the current behavior.

Reply:
Oh man, I was looking at the code in https://github.com/lmorchard/node-firelogger, forgetting that the `firelogger` code was brought into this repo (and I think I was the one that did it, lol)! Sorry! If it was useful for you, I think we should keep it.

Reply:
Oops, I missed your latest comment. Good point, if you're OK with it, let's just use `console.log` instead of `winston`, and then we can dump the dependency on `winston`.

Reply:
Fixed in af2b87f 
Winston is gone!